### PR TITLE
Handle open(filename, 'r')

### LINKED ISSUE #2198: [BUG] Passing non-bytes file input leads to error
https://t.me/pythontelegrambotgroup/396541

TL;DR:

`send_document(open('text_file', 'rb'))` works but `send_document(open('text_file', 'r'))` raises is error.
This is, because we try to guess if the file is an image using `imghdr.what(None, stream)` in `InputFile.is_image`, which only works if `stream` is a bytes stream.
If I comment the `is_image` out, the file is sent without issue, so I guess we should just check if the input is bytes before calling `is_image`

### PR DESCRIPTION
**Pull request by Bibo-Joshi**
closes #2198

### CODE REVIEW

File: telegram/files/inputfile.py [hunk 1]
```diff
@@ -59,9 +59,10 @@ def __init__(self, obj: IO, filename: str = None, attach: bool = None):
         elif hasattr(obj, 'name') and not isinstance(obj.name, int):
             self.filename = os.path.basename(obj.name)
 
-        try:
-            self.mimetype = self.is_image(self.input_file_content)
-        except TelegramError:
+        image_mime_type = self.is_image(self.input_file_content)
+        if image_mime_type:
+            self.mimetype = image_mime_type
+        else:
```
Comment (tsnoam) [inline telegram/files/inputfile.py:65]: elif ...

File: telegram/files/inputfile.py [hunk 2]
```diff
@@ -74,21 +75,27 @@ def field_tuple(self) -> Tuple[str, bytes, str]:
...
-    def is_image(stream: bytes) -> str:
+    def is_image(stream: bytes) -> Optional[str]:
         """Check if the content file is an image by analyzing its headers.
 
         Args:
             stream (:obj:`bytes`): A byte stream representing the content of a file.
 
         Returns:
-            :obj:`str`: The str mime-type of an image.
+            :obj:`str` | :obj:`None`: The mime-type of an image, if the input is an image, and
```
Comment (tsnoam) [inline telegram/files/inputfile.py:85]:
```suggestion
            :obj:`str` | :obj:`None`: The mime-type of an image, if the input is an image, or
```

File: telegram/files/inputfile.py [hunk 3]
```diff
@@ -74,21 +75,27 @@ def field_tuple(self) -> Tuple[str, bytes, str]:
...
-        image = imghdr.what(None, stream)
-        if image:
-            return 'image/%s' % image
-
-        raise TelegramError('Could not parse file content')
+        try:
+            image = imghdr.what(None, stream)
+            if image:
+                return f'image/{image}'
+            return None
```
Comment (tsnoam) [inline telegram/files/inputfile.py:93]: `return None` is redundant. You can remove that line.

File: telegram/files/inputfile.py [hunk 4]
```diff
@@ -74,21 +75,27 @@ def field_tuple(self) -> Tuple[str, bytes, str]:
...
+        except Exception:
+            logger.debug(
+                "Could not parse file content. Assuming that file is not an image.", exc_info=True
+            )
+            return None
```
Comment (tsnoam) [inline telegram/files/inputfile.py:98]: `return None` is redundant. You can remove that line.

**Review by tsnoam — CHANGES_REQUESTED** [review]
Some style nitpicking.
Fix or ignore at your discretion.

File: telegram/files/inputfile.py (hunk 3 above)
Comment (Bibo-Joshi) [reply telegram/files/inputfile.py:93]: mypy & pylint like that …

**Pull request merged**